api/String.cpp

```diff
@@ -652,9 +652,9 @@ void String::replace(const String& find, const String& replace)
 		}
 	} else if (diff < 0) {
 		unsigned int size = len; // compute size needed for result
+        diff = 0 - diff;
```

Comment:
```suggestion
		diff = 0 - diff;
```

Use tabs for indentation to be consistent with the established code style of the file.